src/components/draw/DrawDirective.js

```diff
@@ -37,12 +38,116 @@ goog.require('ga_map_service');
+      var findIconSize = function(olIcon, iconSizes) {
+        var scale = olIcon.getScale();
+        for (var i = 0; i < iconSizes.length; i++) {
+          if (scale == iconSizes[i].scale) {
+            return iconSizes[i];
+          }
+        }
+        return iconSizes[0];
+      };
+
+      var findColor = function(olColor, colors) {
+        var rgb = ol.color.asString(olColor.slice(0, 3));
+        for (var i = 0; i < colors.length; i++) {
+          if (rgb == ol.color.asString(colors[i].fill)) {
+            return colors[i];
+          }
+        }
+        return colors[0];
+      };
+
+      // Creates a new help tooltip
+      var createHelpTooltip = function() {
+        var tooltipElement = document.createElement('div');
+        tooltipElement.className = 'ga-draw-help';
+        var tooltip = new ol.Overlay({
+          element: tooltipElement,
+          offset: [15, 15],
+          positioning: 'top-left',
+          stopEvent: false
+        });
+        return tooltip;
+      };
+
+      // Creates a new measure tooltip with a nice arrow
+      var createMeasureTooltip = function() {
+        var tooltipElement = document.createElement('div');
+        tooltipElement.className = 'ga-draw-measure';
+        var tooltip = new ol.Overlay({
+          element: tooltipElement,
+          offset: [0, -15],
+          positioning: 'bottom-center',
+          stopEvent: true // for copy/paste
+        });
+        return tooltip;
+      };
+
+      var updateHelpTooltip = function(overlay, type, drawStarted, onFirstPoint,
+          onLastPoint) {
+        var helpMsg = $translate.instant('draw_start_' + type);
+        if (drawStarted) {
+          if (type != 'Point') {
+            helpMsg = $translate.instant('draw_next_' + type);
+          }
+          if (onLastPoint) {
+            helpMsg = $translate.instant('draw_snap_last_point_' + type);
+          }
+          if (onFirstPoint) {
+            helpMsg = $translate.instant('draw_snap_first_point_' + type);
+          }
+        }
+        overlay.getElement().innerHTML = helpMsg;
+      };
+
+      // Display measure overlays if the geometry alloow it otherwise hide them.
+      var updateMeasureTooltips = function(distTooltip, areaTooltip, geom) {
+        if (geom instanceof ol.geom.Polygon) {
+          if (areaTooltip) {
+            areaTooltip.getElement().innerHTML = gaMeasure.getAreaLabel(geom);
+            areaTooltip.setPosition(geom.getInteriorPoint().getCoordinates());
+          }
+          geom = new ol.geom.LineString(geom.getCoordinates()[0]);
+        } else if (areaTooltip) {
+          areaTooltip.setPosition(undefined);
+        }
+        if (geom instanceof ol.geom.LineString) {
+          if (distTooltip) {
+            var label = '';
+            var coords = geom.getCoordinates();
+            if (coords.length == 2 ||
+                (coords.length == 3 && coords[1][0] == coords[2][0] &&
+                coords[1][1] == coords[2][1])) {
```

Comment:
Same if clause as in style function. Could be put into utility function. Not sure if worth the effort though.